src/issuance.rs

```diff
@@ -33,6 +33,46 @@ pub struct IssueBundle<T: IssueAuth> {
     authorization: T,
 }
 
+/// Represents the amount of an asset and its finalization status.
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(PartialEq))]
+pub struct AssetSupply {
+    /// The amount of the asset.
+    amount: ValueSum,
+    /// Whether or not the asset is finalized.
+    is_finalized: bool,
+}
```

Comment:
Our new functionality justifies a new file. Please move all data structs and supporting functions/implementations/ tests into `supply_info.rs`.
Leave `verify_supply()` (see below) and verification tests for it in this file.

Reply:
Done.